extensions-jvm/azure-storage-datalake/integration-test/src/test/java/org/apache/camel/quarkus/component/azure/storage/datalake/it/AzureStorageDatalakeTest.java

```diff
@@ -0,0 +1,115 @@
+        // RestAssured.get("/azure-storage-datalake/filesystems")
+        // .then()
+        // .statusCode(200)
+        // .body(Matchers.not(Matchers.hasItem(filesystem)));
+
+        try {
+            /* Create the filesystem */
+            RestAssured.given()
+                    .post("/azure-storage-datalake/filesystem/" + filesystem)
+                    .then()
+                    .statusCode(201);
+
+            /* Now it should exist */
+            // TODO this causes an infinite loop of requests see
+            // https://github.com/apache/camel-quarkus/issues/2304
+            //            RestAssured.get("/azure-storage-datalake/filesystem/" + filesystem)
+            //                    .then()
+            //                    .statusCode(200)
+            //                    .body(Matchers.hasItem(filesystem));
+
+            /* No paths yet */
+            RestAssured.get("/azure-storage-datalake/filesystem/" + filesystem + "/paths")
+                    .then()
+                    .statusCode(200)
+                    .body("", Matchers.hasSize(0));
+
+            String content = "Hello " + RandomStringUtils.randomNumeric(16);
+            /* Upload */
+            RestAssured.given()
+                    .body(content)
+                    .post("/azure-storage-datalake/filesystem/" + filesystem + "/path/" + filename)
+                    .then()
+                    .statusCode(201);
+
+            /* The path occurs in the list */
+            RestAssured.get("/azure-storage-datalake/filesystem/" + filesystem + "/paths")
+                    .then()
+                    .statusCode(200)
+                    .body("", Matchers.hasItem(filename));
+
+            /* Get the file */
+            RestAssured.given()
+                    .get("/azure-storage-datalake/filesystem/" + filesystem + "/path/" + filename)
+                    .then()
+                    .statusCode(200)
+                    // TODO should be Matches.is(content) https://github.com/apache/camel-quarkus/issues/2302
+                    .body(Matchers.startsWith(content));
+
+            /* Consumer */
+            RestAssured.given()
+                    .get("/azure-storage-datalake/consumer/" + filesystem + "/path/" + filename)
+                    .then()
+                    .statusCode(200)
+                    // TODO should be Matches.is(content) https://github.com/apache/camel-quarkus/issues/2302
+                    .body(Matchers.startsWith(content));
+
+        } finally {
+            /* Clean up */
+
+            RestAssured.given()
+                    .delete("/azure-storage-datalake/filesystem/" + filesystem + "/path/" + filename)
+                    .then()
+                    .statusCode(204);
+
+            RestAssured.given()
+                    .delete("/azure-storage-datalake/filesystem/" + filesystem)
+                    .then()
+                    .statusCode(204);
```

Comment:
I wonder if we should assert the deletions here.

Reply:
Yes, we could, although I wonder whether that level of paranoia wouldn't be better suited for the tests in Camel. I think here in Camel Quarkus the main purpose of the tests is to ensure that the main execution paths work flawlessly on Quarkus incl. native mode. If the delete request does not throw an exception, then I think the functionality can be considered proven. But yes, checking the state of the remote storage would be even safer.

Reply:
Indeed, I agree it's not really the scope of camel-quarkus test.